sidecar/src/service/tracing/trace_flusher.rs

```diff
@@ -268,3 +296,169 @@ impl TraceFlusher {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use datadog_trace_protobuf::pb;
+    use datadog_trace_utils::trace_utils::TracerHeaderTags;
+    use httpmock::{Mock, MockServer};
+    use std::sync::Arc;
+
+    // This function will poll the mock server for "hits" until the expected number of hits is
+    // observed. In its current form it may not correctly report if more than the asserted number of
+    // hits occurred. More attempts at lower sleep intervals is preferred to reduce flakiness and
+    // test runtime.
+    async fn poll_for_mock_hit(
```

Comment:
Sleeping during tests are not ideal, but somewhat necessary with the given implementation of the flusher. The public `enqueue` method doesn't ever block on flushing. This function attempts to minimize the badness of sleeping by sleeping for shorter durations, but polling frequently to see if our mocks have been hit.

Reply:
Tried a few ideas but none of them worked without tweaking the flusher. Given the fact that the future inside the flusher is completed in another thread I guess we don't have other choice to just calling await while in the loop.